d3a_interface/device_validator.py

```diff
@@ -60,9 +61,34 @@ def validate_energy(cls, **kwargs):
     @classmethod
     def validate_rate(cls, **kwargs):
         """Validate rates of a Home Meter device."""
+
+        cls._validate_fit_to_limit(
+            fit_to_limit=kwargs.get("fit_to_limit"),
+            energy_rate_increase_per_update=kwargs.get("energy_rate_increase_per_update"),
+            energy_rate_decrease_per_update=kwargs.get("energy_rate_decrease_per_update"))
```

Comment:
it would be better if we stick to a way of calling the validators, either call them and pass the kwargs arguments at once and let them handle unpacking them, or explicitly passing arguments for each validator

Reply:
I think using `**kwargs` to later unpack lots of parameters should be avoided as much as possible. In some cases it can help, but most of the time it creates confusion and can lead to unexpected behavior.
I much prefer, when possible, having named parameters.

In this case, I think it's fine to pass `**kwargs` to the main `validate()` method. Internally, however, the other methods (like `_validate_fit_to_limit`) should make it clear that they are specialized in handling specific parameters.

Reply:
Actually, I believe that `_validate_fit_to_limit` and all other internal validations should not be methods, but rather separate classes (e.g. `FitToLimitValidator`, `EnergyRateValidator` and so on). This would allow combining them together more flexibly. But this might be part of a separate refactoring PR.

Reply:
I agree, it's just that I see in the same scope functions with different arguments style, for example the `_validate_home_meter_consumption_rates` is being passed kwargs arguments